sympy/functions/special/tests/test_bsplines.py

```diff
@@ -100,3 +100,38 @@ def test_repeated_degree_2():
     assert splines[4] == Piecewise((x**2/2 - 2*x + 2, And(x <= 3, x >= 2)),
                                    (-3*x**2/2 + 10*x - 16, And(x <= 4, x >= 3)),
                                    (0, True))
+
+# Tests for interpolating_spline
+
+
+def test_10_points_degree_1():
+    d = 1
+    X = [-5, 2, 3, 4, 7, 9, 10, 30, 31, 34]
+    Y = [-10, -2, 2, 4, 7, 6, 20, 45, 19, 25]
+    spline = interpolating_spline(d, x, X, Y)
+
+    assert spline == Piecewise((8*x/7 - 30/7, (x >= -5) & (x <= 2)), (4*x - 10, (x >= 2) & (x <= 3)),
+                               (2*x - 4, (x >= 3) & (x <= 4)), (x, (x >= 4) & (x <= 7)),
+                               (-x/2 + 21/2, (x >= 7) & (x <= 9)), (14*x - 120, (x >= 9) & (x <= 10)),
+                               (5*x/4 + 15/2, (x >= 10) & (x <= 30)), (-26*x + 825, (x >= 30) & (x <= 31)),
+                               (2*x - 43, (x >= 31) & (x <= 34)))
+
+
+def test_3_points_degree_2():
```

Comment:
In testing, I found the function fails for degree 2 with more than 3 points, because this case uses Rational which isn't imported. On top of bspline.py, 
```
from sympy.core import S, sympify
```
should be
```
from sympy.core import S, sympify, Rational
```

And to make sure this is fixed, please add a test for degree 2 with 5 points.